rust-keybroker/keybroker-server/src/main.rs

```diff
@@ -139,19 +141,37 @@ async fn submit_evidence(
 #[derive(Clone, Parser, Debug)]
 #[command(version, about, long_about = None)]
 struct Args {
-    /// The port on which the web server will listen
+    /// The interface on which this server will listen (use 0.0.0.0 to listen on all interfaces)
+    #[arg(short, long, default_value = "127.0.0.1")]
+    addr: String,
+
+    /// The port on which this server will listen
     #[arg(short, long, default_value_t = 8088)]
     port: u16,
 
-    #[arg(short, long, default_value = "http://127.0.0.1")]
-    baseurl: String,
+    /// The base URL at which this server can be reached back for evidence submission.
+    /// If not specified on the command line, it will be set to 'http://{addr}', but
+    /// this value can be overridden with an FQDN for {addr} in order to use name resolution.
+    /// The port number will be appended, so don't leave a trailing '/' to the URL.
+    #[arg(short, long, default_value = None)]
+    base_url: Option<String>,
```

Comment:
It seems more uniform to require a single `base_url` argument (maybe `origin` would be more precise, including an explicit scheme and authority part, e.g.:  http://example.com:8042), rather than also allowing `addr` and `port`. I think it could be a source of confusion UX-wise.  Then we rely on a URL-parsing library to extract the address and port (and in the future, the scheme too in case we want to support HTTPS.)

Reply:
The server is _special_ in the sense that you need to take into account 2 different --- but related --- things. The `addr` and `port` parameters do relate to the physical network interfaces the server will listen on (e.g. eth0), whereas `base_url` is FQDN the server can be reached at.

Reply:
Is the `base_url` the name that would appear in the [`Host` header](https://stackoverflow.com/questions/43156023/what-is-http-host-header) in HTTP requests to the server?

If so, maybe we could call it `virtual_host` or `vhost` or similar.

Reply:
I agree that using the proper option name is important as it implies some representation in the user's mind. Here, this is really just an FQDN. I believe (I'm not an expert) that this is part of the Host header in HTTP, but it is not intentional. At this stage, it's just a name that allows DNS / load balancing / ... I find that `virtual_host` / `vhost` are very much loaded with some web server aspect. Ultimately, in a production environment, it probably will have to, but for now, it corresponds to the `--endpoint` that is used on the `keybroker-app` side of things. Lacking any better term, could not just we reuse the `--endpoint <ENDPOINT>` as well on the server side ? This would echo the client side (which is good for the least surprise of our users), and just mean the server exposes this endpoint on the network (without implying any specific implementation details).

Reply:
Sure, my primary concern was to avoid "base URL" - which comes loaded with a pretty precise meaning.  I am OK with the "endpoint" suggestion you make above.

Reply:
The latest patch now makes use of endpoint !